src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,89 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTORY = 0.75f;
+    private Node<K, V>[] table;
+    private int capacity;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        capacity = INITIAL_CAPACITY;
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTORY);
+    }
+
+    public static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size == threshold) {
+            resize();
+        }
+        Node<K, V> node = new Node<>(key, value, null);
+        Node<K, V> currentNode = table[getIndex(key)];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = node;
+                size++;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        table[getIndex(key)] = node;
+        size++;
+    }
+
+    private void resize() {
+        capacity = capacity * 2;
+        threshold = (int) (capacity * LOAD_FACTORY);
+        size = 0;
 
+        Node<K, V>[] oldTable = table;
+        table = new Node[capacity];
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
     }
```

Comment:
private methods should follow after public as helper methods

Reply:
fixed